RileyLinkKit/PumpOpsSynchronous.swift

```diff
@@ -468,18 +468,24 @@ class PumpOpsSynchronous {
                 timestamp.timeZone = pump.timeZone
 
                 if let date = timestamp.date?.addingTimeInterval(timeAdjustmentInterval) {
-                    if date.timeIntervalSince(startDate) < -eventTimestampDeltaAllowance {
-                        NSLog("Found event at (%@) to be more than %@s before startDate(%@)", date as NSDate, String(describing: eventTimestampDeltaAllowance), startDate as NSDate);
-                        return (events: events, hasMoreEvents: false)
-                    } else if date.timeIntervalSince(timeCursor) > eventTimestampDeltaAllowance {
-                        NSLog("Found event (%@) out of order in history. Ending history fetch.", date as NSDate)
-                        return (events: events, hasMoreEvents: false)
-                    } else {
-                        if (date.compare(startDate) != .orderedAscending) {
-                            timeCursor = date
+                    
+                    let dontCancelBecauseOfDate = pumpModel.mayHaveOutOfOrderEvents && event.canBeDelayedAppend()
+                    
+                    if !dontCancelBecauseOfDate {
+                        if date.timeIntervalSince(startDate) < -eventTimestampDeltaAllowance {
+                            NSLog("Found event at (%@) to be more than %@s before startDate(%@)", date as NSDate, String(describing: eventTimestampDeltaAllowance), startDate as NSDate);
+                            return (events: events, hasMoreEvents: false)
+                        } else if date.timeIntervalSince(timeCursor) > eventTimestampDeltaAllowance {
+                            NSLog("Found event (%@) out of order in history. Ending history fetch.", date as NSDate)
+                            return (events: events, hasMoreEvents: false)
```

Comment:
yes, either an error or a flag indicating loss of interpretability of pump event time.